inst/include/population_dynamics/fleet/fleet.hpp

```diff
@@ -1,43 +1,100 @@
- * do not and cannot warrant the performance or results that may be obtained by
- * using this  software or data. The National Oceanic And Atmospheric
- * Administration and the U.S. Government disclaim all warranties, express or
- * implied, including warranties of performance, merchantability or fitness
- * for any particular purpose.
- *
- * Please cite the author(s) in any work or product based on this material.
- *
+ * The purpose of this file is to declare the growth functor class
+ * which is the base class for all growth functors.
  */
 #ifndef FIMS_POPULATION_DYNAMICS_FLEET_HPP
 #define FIMS_POPULATION_DYNAMICS_FLEET_HPP
 
 #include "../../common/model_object.hpp"
+#include "../../common/data_object.hpp"
+#include "../../distributions/distributions.hpp"
+#include "../selectivity/selectivity.hpp"
 
 namespace fims {
 
-template <typename T>
-struct Fleet {};
+    /* @brief Base class for all fleets.
+     *
+     * @tparam Type The type of the fleet object.
+     * */
+    template<class Type>
+    struct Fleet : public FIMSObject<Type> {
+        static uint32_t id_g; /*!< reference id for fleet object*/
+        size_t nyears; /*!< the number of years in the model*/
+        size_t nages; /*!< the number of ages in the model*/
+
+
+        //data objects
+        int observed_catch_data_id = -999; /*!< id of observed catch data object*/
+        std::shared_ptr<fims::DataObject<Type> > observed_catch_data; /*!< observed catch data object*/
+
+        int observed_index_data_id = -999; /*!< id of observed index data object*/
+        std::shared_ptr<fims::DataObject<Type> > observed_index_data; /*!< observed index data object*/
+
+        int observed_agecomp_data_id = -999; /*!< id of observed agecomp data object*/
+        std::shared_ptr<fims::DataObject<Type> > observed_agecomp_data; /*!< observed agecomp data object*/
+
+        //likelihood components
+        int catch_likelihood_id = -999; /*!< id of catch likelihood component*/
+        std::shared_ptr<fims::DistributionsBase<Type> > catch_likelihood; /*!< catch likelihood component*/
+
+        int index_likelihood_id = -999; /*!< id of index likelihood component*/
+        std::shared_ptr<fims::DistributionsBase<Type> > index_likelihood; /*!< index likelihood component*/
+
+        int agecomp_likelihood_id = -999; /*!< id of agecomp likelihood component*/
+        std::shared_ptr<fims::DistributionsBase<Type> > agecomp_likelihood; /*!< agecomp likelihood component*/
+
+        //selectivity
+        int selectivity_id = -999;  /*!< id of selectivity component*/
+        std::shared_ptr<fims::SelectivityBase<Type> > selectivity; /*!< selectivity component*/
+
+        //derived quantities
+        std::vector<Type> expected_catch; /*!<model expected total catch*/
+        std::vector<Type> expected_index; /*!<model expected index of abundance*/
+        std::vector<Type> catch_numbers_at_age; /*!<model expected catch at age*/
+        
```

Comment:
need agecomp vector

Reply:
Here the catch_numbers_at_age is the comp vector. We did have discussion regarding when comps should be whole numbers or proportions so should that scaling be done by the user or inside FIMS